sharedbuffers/mapped_struct.py

```diff
@@ -812,14 +817,37 @@ def unpack_from(cls, buf, offs, idmap = None):
         buf = _likerobuffer(buf)
         return cls(buf, offs, idmap)
 
-    @cython.locals(obj_offs = cython.ulonglong, dcode = cython.char, index = cython.ulonglong, pindex = "unsigned long *", dataoffs = cython.ulonglong)
-    def __getitem__(self, index):
+    def _getslice(self, start, end, step):
+        if step < 0:
+            while end >= start:
+                yield self[end + step]
+                end += step
+        else:
+            while start < end:
+                yield self[start]
+                start += step
```

Comment:
No, a generator is not the idea. The idea is to returned a mapped_list that just points to a different slice of the buffer. Maybe you can play with strides to be able to support steps without much hardship.